mbedtls/src/bignum/mod.rs

```diff
@@ -133,22 +133,23 @@ impl Mpi {
         Ok(())
     }
 
-    fn get_limb(&self, n: usize) -> mpi_uint {
-        if n < self.inner.n {
-            unsafe { *self.inner.p.offset(n as isize) }
-        } else {
-            // zero pad
-            0
-        }
-    }
-
     pub fn as_u32(&self) -> Result<u32> {
         if self.bit_length()? > 32 {
             // Not exactly correct but close enough
             return Err(codes::MpiBufferTooSmall.into());
         }
 
-        Ok(self.get_limb(0) as u32)
+        let mut buf = [0u8; 4];
+        if cfg!(target_endian = "little")
```

Comment:
Why do you need this check? can't you use either `mpi_write_binary_le/u32::from_le_bytes` or `mpi_write_binary/u32::from_be_bytes` regardless of target?